roles/neutron-common/templates/usr/local/bin/neutron-reset

```diff
@@ -61,33 +61,56 @@ class Interface(object):
             return True
 
         # veth interfaces named tap... belong to Neutron
-        if re.match(r'^tap[0-9a-z]{8}-[0-9a-z]{2}$', self.ifname) and \
-                self.iftype() == 'veth':
+        if re.match(r'^tap[0-9a-z]{8}-[0-9a-z]{2}$', self.ifname) \
+                and self.iftype() == 'veth':
             return True
 
-        if re.match(r'^vxlan[\d]{1-4}$', self.ifname) and \
-                self.iftype() == 'vxlan':
+        if re.match(r'^vxlan\-[0-9]{1,4}$', self.ifname) \
+                and self.iftype() == 'vxlan':
             return True
 
-        if re.match(r'^brq[0-9a-z]{8}-[0-9a-z]{2}$', self.ifname) and \
-                self.iftype() == 'bridge':
+        if re.match(r'^brq[0-9a-z]{8}-[0-9a-z]{2}$', self.ifname) \
+                and self.iftype() == 'bridge':
             return True
 
         return False
 
+    def is_nova_interface(self):
+        return self.ifname.startswith('tap') and self.iftype() == 'tun'
+
     def iftype(self):
         if self._iftype is None:
             self._fetch_additional_info()
 
         return self._iftype
 
+    def attached_bridge(self):
+        if self._bridge_attached is None:
+            self._fetch_additional_info()
+
+        return self._bridge_attached
+
+    def wait_for_link_up(self):
+        cmd = self._ip_cmd() + ['link', 'show', self.ifname]
+
+        while True:
+            try:
+                output = subprocess.check_output(cmd)
+
+                for line in output.splitlines():
+                    if self.ifname in line and "state UP" in line:
+                        return
+
```

Comment:
What about the case that `ip link show foodev` returns an exit code of 0, but it doesn't contain 'link up'? As the code is now, it will busy loop rather than sleeping.